librad/src/git/storage.rs

```diff
@@ -554,11 +554,35 @@ impl Storage {
         T: Serialize + DeserializeOwned + Clone + EntityInfoExt,
     {
         self.entity_metadata_commit(urn).and_then(|commit| {
-            Entity::<T, Draft>::from_json_slice(self.entity_blob(commit).unwrap().content())
-                .map_err(Error::Entity)
+            Ok(Entity::<T, Draft>::from_json_slice(
+                self.entity_blob(commit)?.content(),
+            )?)
         })
     }
 
+    /// Get all the [`Entity`] data in this `Storage`.
+    ///
+    /// The caller has the choice to filter on the [`EntityInfo`], which is
+    /// useful when the you want a list of a specific kind of `Entity`.
+    pub fn entities(&self) -> Result<Vec<GenericDraftEntity>, Error> {
+        let references = self.backend.references_glob("refs/namespaces/**/rad/id")?;
```

Comment:
Why did you choose to glob on `**`? Iiuc, this would also match `rad/id` branches of remotes. Those may or may not be verified, and may or may not agree. In the best case, the output would include duplicate `GenericDraftEntity` values.

Reply:
Ah you're right. I messed up the globbing in my head. What I want is `refs/namespace/*/rad/id` to get all the first level entities. I can submit a patch. Sorry for the haste.